src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,114 @@
+            throw new DukeException("Exception while opening task list file: " + e);
+        }
+
+        ArrayList<Task> outputTaskList = new ArrayList<>();
+        Scanner taskReader;
+        try {
+            taskReader = new Scanner(taskListFile);
+            while (taskReader.hasNextLine()) {
+                String taskFromFile = taskReader.nextLine();
+                // Note, this is assuming that format of
+                // task.Task.getDescriptionForDatabase() remains the same.
+                String[] formattedTaskString = taskFromFile.split(" - ");
+                Parser.Command taskCommand = Parser.parseCommand(formattedTaskString[0]);
+                boolean isTaskDone = formattedTaskString[1].equals("1");
+                switch (taskCommand) {
+                case TODO:
+                    outputTaskList.add(new Todo(formattedTaskString[2], isTaskDone));
+                    break;
+                case EVENT:
+                    outputTaskList.add(
+                            new Event(
+                                    formattedTaskString[2],
+                                    formattedTaskString[3],
+                                    isTaskDone
+                            )
+                    );
+                    break;
+                case DEADLINE:
+                    outputTaskList.add(
+                            new Deadline(
+                                    formattedTaskString[2],
+                                    formattedTaskString[3],
+                                    isTaskDone
+                            )
+                    );
+                    break;
+                }
+            }
+            taskReader.close();
+        } catch (FileNotFoundException e) {
+            throw new DukeException("Exception while scanning task list file: " + e);
+        }
+        return outputTaskList;
+    }
+
+    /**
+     * Saves the Tasks stored in a TaskList into the file path of the Storage.
+     * @param taskList TaskList object containing Task(s) to store.
+     * @return true if store was successful.
+     * @throws DukeException Exception while storing into file.
+     */
+    boolean store(TaskList taskList) throws DukeException {
+        try {
+            File taskListFile = new File(filePath);
+            // Clear the pre-existing file if there is one.
+            if (taskListFile.exists()) {
+                taskListFile.delete();
+                taskListFile.createNewFile();
+            } else {
+                throw new IOException("No file found at the specified path.");
+            }
+            if (taskList.getSize() > 0) {
+                FileWriter taskWriter = new FileWriter(taskListFile);
+                for (Task t : taskList.getTaskList()) {
+                    taskWriter.write(t.getDescriptionForDatabase());
+                    taskWriter.write("\n");
+                }
+                taskWriter.close();
+            }
```

Comment:
Consider separating different logical units with a line break.

Reply:
Thanks for the comments Jian Han 👍